ansible_base/authentication/authenticator_plugins/ldap.py

```diff
@@ -372,19 +436,28 @@ def authenticate(self, request, username=None, password=None, **kwargs) -> (obje
             self.settings.CONNECTION_OPTIONS[ldap.OPT_X_TLS_NEWCTX] = 0
 
         # Ensure USER_SEARCH and GROUP_SEARCH are converted into a search object
-        for field, search_must_have_user in [('GROUP_SEARCH', False), ('USER_SEARCH', True)]:
+        for field in ['GROUP_SEARCH', 'USER_SEARCH']:
             data = getattr(self.settings, field, None)
             # Ignore None or empty (e.g., [])
             if not data:
                 setattr(self.settings, field, None)
-            elif not isinstance(data, config.LDAPSearch):
-                try:
-                    # Search fields should be LDAPSearch objects, so we need to convert them from [] to these objects
-                    search_object = config.LDAPSearch(data[0], getattr(ldap, data[1]), data[2])
-                    setattr(self.settings, field, search_object)
-                except Exception as e:
-                    logger.error(f'Failed to instantiate {field} LDAPSearch object: {e}')
-                    return None
+            elif not isinstance(data, config.LDAPSearch) and not isinstance(data, config.LDAPSearchUnion):
+                # If we only got 3 strings we will be just a single search, stuff it into an array of a single search
+                if len(data) == 3 and type(data[0]) and type(data[1]) is str and type(data[2]) is str:
```

Comment:
type(data[0]) missing "is str"?  Or what is a truthy type?

Reply:
What if data = "ABC"? :)

Reply:
Nice catch, thanks!